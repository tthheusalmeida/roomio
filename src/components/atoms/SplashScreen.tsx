import Loading from "./Loading";

export default function SplashScreen() {
  return (
    <div className="h-dvh w-dvw flex items-center justify-center bg-violet-900">
      <Loading height={120} width={120} />
    </div>
  );
}
