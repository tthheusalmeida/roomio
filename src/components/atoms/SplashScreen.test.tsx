import { render, screen } from "@testing-library/react";
import SplashScreen from "./SplashScreen";

jest.mock("./LoadingLogo", () => ({
  __esModule: true,
  default: () => <div data-testid="loading-logo" />,
}));

describe("SplashScreen component", () => {
  it("renders the container and LoadingLogo", () => {
    const { asFragment } = render(<SplashScreen />);

    const container = screen.getByTestId("loading-logo").parentElement;
    expect(container).toHaveClass(
      "h-dvh",
      "w-dvw",
      "flex",
      "items-center",
      "justify-center",
      "bg-violet-900"
    );

    expect(screen.getByTestId("loading-logo")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
