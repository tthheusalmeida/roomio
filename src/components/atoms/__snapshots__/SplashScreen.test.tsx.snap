// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SplashScreen component renders the container and LoadingLogo 1`] = `
<DocumentFragment>
  <div
    class="h-dvh w-dvw flex items-center justify-center bg-violet-900"
  >
    <div
      data-testid="loading-logo"
    />
  </div>
</DocumentFragment>
`;
